verifcid/cid.go

```diff
@@ -0,0 +1,33 @@
+package verifcid
+
+import (
+	"fmt"
+
+	"github.com/ipfs/go-cid"
+	mh "github.com/multiformats/go-multihash"
+)
+
+var ErrPossiblyInsecureHashFunction = fmt.Errorf("potentially insecure hash functions not allowed")
+var ErrBelowMinimumHashLength = fmt.Errorf("hashes must be at least %d bytes long", minimumHashLength)
+var ErrAboveMaximumHashLength = fmt.Errorf("hashes must be at most %d bytes long", maximumHashLength)
+
+const minimumHashLength = 20
+const maximumHashLength = 128
+
+// ValidateCid validates multihash allowance behind given CID.
+func ValidateCid(allowlist Allowlist, c cid.Cid) error {
```

Comment:
Could we make this non breaking, better name than `ValidateCidWithAllowlist` wanted tho ?
```suggestion
// ValidateCid validates multihash allowance behind given CID.
// It uses [DefaultAllowlist].
func ValidateCid(c cid.Cid) error {
 return ValidateCidWithAllowlist(DefautAllowlist, c)
}

// ValidateCidWithAllowlist is like [ValidateCid] but it uses a custom [Allowlist] instead of [DefaultAllowlist].
func ValidateCidWithAllowlist(allowlist Allowlist, c cid.Cid) error {
```

Reply:
What's the point? I thought we are breaking stuff here. 

The only users of the default allowlist are supposed to be boxo/kubo libs, but now they enable customisable allowlists and will always use ValidateWithAllowlist.

Who is going to use ValidateCid with default?

Reply:
Ah right, I remember we talked about that, this is good.
I forgot no one is using this code.